jparest/src/main/java/uk/gov/homeoffice/digital/sas/jparest/validation/EntityValidator.java

```diff
@@ -36,25 +40,31 @@ public void validateAndThrowIfErrorsExist(Object objectToValidate) {
 
       if (!constraintViolations.isEmpty()) {
         throw new ResourceConstraintViolationException(
-            createResourceConstraintViolationMessage(constraintViolations));
+            createStructuredErrors(constraintViolations));
       }
     }
   }
 
-  private static String createResourceConstraintViolationMessage(
-      Set<ConstraintViolation<Object>> constraintViolations) {
+  private static List<StructuredError> createStructuredErrors(
+          Set<ConstraintViolation<Object>> constraintViolations) {
+
+    var constraintViolation = constraintViolations.iterator().next();
+    var hibernateConstraintViolation = constraintViolation.unwrap(
+        HibernateConstraintViolation.class
+    );
+    var payload = hibernateConstraintViolation.getDynamicPayload(ArrayList.class);
```

Comment:
Do we need to store this within a variable? can we put the getDynamicPayload directly into the return statement?

Reply:
👍  done